src/v/cluster/partition_manager.cc

```diff
@@ -162,10 +163,26 @@ ss::future<> partition_manager::stop_partitions() {
     co_await _gate.close();
     // prevent partitions from being accessed
     auto partitions = std::exchange(_ntp_table, {});
-    _raft_table.clear();
+
+    {
+        auto current = _raft_table.begin();
+        while (current != _raft_table.end()) {
+            current = _raft_table.erase(current, ++current);
+            co_await ss::coroutine::maybe_yield();
```

Comment:
I wonder what the order-of-magnitude cost of calling this for every element is. If we had a helper it could do N elements before doing a yield (though you'd need an estimate from the call about how expensive every deletion is to get N right).